src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -766,11 +778,112 @@ public async Task TestPartialProjectSync_Options1()
+
+            // Do the initial full sync
+            await solution.AppendAssetMapAsync(map, CancellationToken.None);
+            var solutionChecksum = await solution.CompilationState.GetChecksumAsync(CancellationToken.None);
+            var fullSyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, solutionChecksum, updatePrimaryBranch: true, workspaceVersion: solution.WorkspaceVersion, CancellationToken.None);
+            Assert.Equal(2, fullSyncedSolution.Projects.Count());
+
+            // Mutate both projects to each have a document in it.
+            solution = solution.GetProject(project1.Id).AddDocument("X.cs", SourceText.From("// X")).Project.Solution;
+            solution = solution.GetProject(project2.Id).AddDocument("Y.vb", SourceText.From("' Y")).Project.Solution;
+
+            // Now just sync project1's cone over.  We should not see the change to project2 on the remote side.
+            // But we will still see project2.
+            {
+                await solution.AppendAssetMapAsync(map, project1.Id, CancellationToken.None);
+                var project1Checksum = await solution.CompilationState.GetChecksumAsync(project1.Id, CancellationToken.None);
+                var project1SyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, project1Checksum, updatePrimaryBranch: false, workspaceVersion: -1, CancellationToken.None);
+                Assert.Equal(2, project1SyncedSolution.Projects.Count());
+                var csharpProject = project1SyncedSolution.Projects.Single(p => p.Language == LanguageNames.CSharp);
+                var vbProject = project1SyncedSolution.Projects.Single(p => p.Language == LanguageNames.VisualBasic);
+                Assert.True(csharpProject.DocumentIds.Count == 2);
+                Assert.True(vbProject.DocumentIds.Count == 0);
+            }
+
+            // Similarly, if we sync just project2's cone over:
+            {
+                await solution.AppendAssetMapAsync(map, project2.Id, CancellationToken.None);
+                var project2Checksum = await solution.CompilationState.GetChecksumAsync(project2.Id, CancellationToken.None);
+                var project2SyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, project2Checksum, updatePrimaryBranch: false, workspaceVersion: -1, CancellationToken.None);
+                Assert.Equal(2, project2SyncedSolution.Projects.Count());
+                var csharpProject = project2SyncedSolution.Projects.Single(p => p.Language == LanguageNames.CSharp);
+                var vbProject = project2SyncedSolution.Projects.Single(p => p.Language == LanguageNames.VisualBasic);
+                Assert.True(csharpProject.DocumentIds.Count == 1);
+                Assert.True(vbProject.DocumentIds.Count == 1);
+            }
+        }
+
+        [Fact]
+        public async Task TestPartialProjectSync_AddP2PRef()
```

Comment:
this was an interesting case that helped motivate the larger scale change in SolutionCreator.